src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/EventHub/RemoveAzureEventHub.cs

```diff
@@ -14,35 +14,71 @@
+        [Parameter(Mandatory = true, ParameterSetName = EventhubDefaultParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name")]
         [ResourceGroupCompleter]
         [ValidateNotNullOrEmpty]
          public string ResourceGroupName { get; set; }
 
-        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Namespace Name")]
+        [Parameter(Mandatory = true, ParameterSetName = EventhubDefaultParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Namespace Name")]
         [ValidateNotNullOrEmpty]
         [Alias(AliasNamespaceName)]
         public string Namespace { get; set; }
 
-        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = "EventHub Name")]
+        [Parameter(Mandatory = true, ParameterSetName = EventhubDefaultParameterSet, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = "EventHub Name")]
         [ValidateNotNullOrEmpty]
         [Alias(AliasEventHubName)]
         public string Name { get; set; }
 
+        [Parameter(Mandatory = true, ParameterSetName = EventhubInputObjectParameterSet, ValueFromPipeline = true, Position = 0, HelpMessage = "Eventhub Object")]
+        [ValidateNotNullOrEmpty]
+        public PSEventHubAttributes InputObject { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = EventhubResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Eventhub Resource Id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
         public override void ExecuteCmdlet()
         {
+
+            if (ParameterSetName.Equals(EventhubInputObjectParameterSet))
+            {
+                LocalResourceIdentifier identifier = new LocalResourceIdentifier(InputObject.Id);
+                ResourceGroupName = identifier.ResourceGroupName;
+                Namespace = identifier.ParentResource;
+                Name = identifier.ResourceName;
+            }
+            else if (ParameterSetName.Equals(EventhubResourceIdParameterSet))
+            {
+                LocalResourceIdentifier identifier = new LocalResourceIdentifier(ResourceId);
+                ResourceGroupName = identifier.ResourceGroupName;
+                Namespace = identifier.ParentResource;
+                Name = identifier.ResourceName;
+            }
+
             // delete a EventHub 
             if(ShouldProcess(target:Name, action:string.Format(Resources.RemovingEventHub,Name,Namespace)))
             {
-                WriteObject(Client.DeleteEventHub(ResourceGroupName, Namespace, Name));
```

Comment:
@v-Ajnava same comment here about the type being returned

Reply:
resolved